src/msw/webview_edge.cpp

```diff
@@ -0,0 +1,777 @@
+                }
+                return S_OK;
+            })
+        .Get(), &m_navigationCompletedToken);
+    m_webView->add_NewWindowRequested(
+        Callback<IWebView2NewWindowRequestedEventHandler>(
+            [this](IWebView2WebView* WXUNUSED(sender), IWebView2NewWindowRequestedEventArgs* args) -> HRESULT
+            {
+                PWSTR uri;
+                args->get_Uri(&uri);
+                wxString evtURL(uri);
+                AddPendingEvent(wxWebViewEvent(wxEVT_WEBVIEW_NEWWINDOW, GetId(), evtURL, wxString()));
+                args->put_Handled(true);
+                return S_OK;
+            }).Get(), &m_newWindowRequestedToken);
+    LoadURL(m_pendingURL);
+}
+
+void wxWebViewEdge::OnSize(wxSizeEvent& event)
+{
+    UpdateBounds();
+    event.Skip();
+}
+
+void wxWebViewEdge::UpdateBounds()
+{
+    RECT r;
+    wxCopyRectToRECT(GetClientRect(), r);
+    if (m_webView)
+        m_webView->put_Bounds(r);
+}
+
+void wxWebViewEdge::LoadURL(const wxString& url)
+{
+    if (!m_webView)
+    {
+        m_pendingURL = url;
+        return;
+    }
+    if (FAILED(m_webView->Navigate(url.wc_str())))
+        wxLogError("Could not navigate to URL");
+}
+
+void wxWebViewEdge::LoadHistoryItem(wxSharedPtr<wxWebViewHistoryItem> item)
+{
+    int pos = -1;
+    for (unsigned int i = 0; i < m_historyList.size(); i++)
+    {
+        //We compare the actual pointers to find the correct item
+        if (m_historyList[i].get() == item.get())
+            pos = i;
+    }
+    // TODO: wxASSERT_MSG(pos != static_cast<int>(m_impl->m_historyList.size()), "invalid history item");
+    m_historyLoadingFromList = true;
+    LoadURL(item->GetUrl());
+    m_historyPosition = pos;
+}
+
+wxVector<wxSharedPtr<wxWebViewHistoryItem> > wxWebViewEdge::GetBackwardHistory()
+{
+    wxVector<wxSharedPtr<wxWebViewHistoryItem> > backhist;
+    //As we don't have std::copy or an iterator constructor in the wxwidgets
+    //native vector we construct it by hand
+    for (int i = 0; i < m_historyPosition; i++)
+    {
+        backhist.push_back(m_historyList[i]);
+    }
+    return backhist;
+}
+
+wxVector<wxSharedPtr<wxWebViewHistoryItem> > wxWebViewEdge::GetForwardHistory()
+{
+    wxVector<wxSharedPtr<wxWebViewHistoryItem> > forwardhist;
+    //As we don't have std::copy or an iterator constructor in the wxwidgets
+    //native vector we construct it by hand
+    for (int i = m_historyPosition + 1; i < static_cast<int>(m_historyList.size()); i++)
+    {
+        forwardhist.push_back(m_historyList[i]);
+    }
+    return forwardhist;
+}
+
+bool wxWebViewEdge::CanGoForward() const
+{
+    if (m_historyEnabled)
+        return m_historyPosition != static_cast<int>(m_historyList.size()) - 1;
+    else
+        return false;
+
+    /*
+        BOOL result = false;
+        if (m_webView && SUCCEEDED(m_webView->get_CanGoForward(&result)))
+            return result;
+        else
+            return false;
+    */
```

Comment:
Do we really need to keep this? If so, a comment explaining why is this code not used but still kept would be helpful.